src/shared/components/oncoprint/DeltaUtils.ts

```diff
@@ -870,28 +912,62 @@ function transitionHeatmapTrack(
             )
         };
-        const newTrackId = oncoprint.addTracks([heatmapTrackParams])[0];
+        // register new track in oncoprint
+        const newTrackId:number = oncoprint.addTracks([heatmapTrackParams])[0];
+        // store relation between React heatmap track specs and OncoprintJS trackIds
         trackSpecKeyToTrackId[nextSpec.key] = newTrackId;
 
-        let trackIdForRuleSetSharingKey:"heatmap"|"heatmap01" = "heatmap";
-        if (nextSpec.molecularAlterationType === "METHYLATION") {
-            trackIdForRuleSetSharingKey = "heatmap01";
+        let trackIdForRuleSetSharingKey:"heatmap"|"heatmap01"|undefined;
+
+        switch (nextSpec.molecularAlterationType) {
+            case AlterationTypeConstants.TREATMENT_RESPONSE:
+
+                trackIdForRuleSetSharingKey = undefined;
+        
+                // register the new track id so that it will determine formatting of the track group
+                const rulesetTrackId = trackIdForRuleSetSharing.treatment![nextSpec.molecularProfileId];
+                if (rulesetTrackId) {
+                    oncoprint.shareRuleSet(newTrackId, rulesetTrackId);
+                }
+
+                break;
+
+            case AlterationTypeConstants.METHYLATION:
+                trackIdForRuleSetSharingKey = "heatmap01"; 
+                break;
+        
+            default:
+                trackIdForRuleSetSharingKey = "heatmap";
+                break;
         }
-        if (typeof trackIdForRuleSetSharing[trackIdForRuleSetSharingKey] !== "undefined") {
-            oncoprint.shareRuleSet(trackIdForRuleSetSharing[trackIdForRuleSetSharingKey]!, newTrackId);
+
+        if (trackIdForRuleSetSharingKey) {
+            if (trackIdForRuleSetSharing[trackIdForRuleSetSharingKey]) {
+                // register the new track id so that its is formatted by the existing track group
+                oncoprint.shareRuleSet(trackIdForRuleSetSharing[trackIdForRuleSetSharingKey]!, newTrackId);
+            }
+            trackIdForRuleSetSharing[trackIdForRuleSetSharingKey] = newTrackId;
         }
-        trackIdForRuleSetSharing[trackIdForRuleSetSharingKey] = newTrackId;
+
     } else if (nextSpec && prevSpec) {
         // Transition track
         const trackId = trackSpecKeyToTrackId[nextSpec.key];
+        // when the data in the next track differs from the previous
+        // register the new data points in oncoprint
         if (nextSpec.data !== prevSpec.data) {
             // shallow equality check
             oncoprint.setTrackData(trackId, nextSpec.data, "uid");
         }
         if (nextSpec.info !== prevSpec.info && nextSpec.info !== undefined) {
             oncoprint.setTrackInfo(trackId, nextSpec.info);
         }
+        // re-register the shared rule set
+        const rulesetTrackId = trackIdForRuleSetSharing.treatment![nextSpec.molecularProfileId];
+        if (rulesetTrackId) {
+            oncoprint.shareRuleSet(rulesetTrackId, trackId);
```

Comment:
(1) why?
(2) this code applies for gene heatmaps as well - methylation, mrna, protein, so it's going to break non-treatment heatmaps

Reply:
(see comment above)

> Ah yes, you are right. This reverse argument does not do what I thought it would do. I would like to point out that there is a reverse logic for treatment profile tracks compared to others. I have looked at the code again and refactored to make this logic more clear. In essence the difference is:
> 
> molecular profiles: a new molecular profile track is linked with the rule set of an existing track via the trackid of the existing set. If its the first track it is added to trackIdForRuleSetSharing so that others can link to it. The ruleSet association of existing tracks is never updated (always link to the first track that was added).
> treatment profiles: a new treatment profile track is always added to trackIdForRuleSetSharing so that others can link to it. The ruleSet association of existing treatment tracks is always updated to the track that was added last.
> The reason for this different logic is that legends/ruleSets for treatments are based on an aggregation of data in all tracks in the trackGroup (data is not standardized f.i. between -3 and 3 Z-scores). The last added track holds up-to-date information on all tracks in the group (includes information on itself). When this track is added the styling of all existing tracks are updated to point to the ruleSet defined by the last added track.
> 
> I think that this reverse logic in not necessary. When all track types would link to the ruleSet of the last added track irrespective of the data type this logic can be removed and the functionality is identical.
> 
> What do you think?